robottelo/hosts.py

```diff
@@ -479,6 +479,118 @@ def install_cockpit(self):
         if result.status != 0:
             raise ContentHostError('Failed to install the cockpit')
 
+    def register(
+        self,
+        target=settings.server.hostname,
```

Comment:
This should be a required parameter, and should not default to the settings hostname.

Satellite/Capsule instance should be passed, and the `target.hostname` attribute used where necessary within the function.